### PR TITLE
Feature specs

### PR DESCRIPTION
**Pull request by sauloperez**
El viernes estuve trabajando en un pequeño refactor de transferencias de tiempo entre miembros y organizaciones y me vi constantemente probando las transferencias manualmente. Aunque tenemos alguno tests justamente tuve que añadir tests de controller para los actions que estoy tocando (pull request en breve).

Sin embargo, para ir más ágil pensé en automitzar el test manual de transferir y así avanzar más rápido en el refactor sin romper nada. Esto al darnos más seguridad de que las cosas no se rompen facilitará las contribuciones esporádicas y quitará algo de trabajo a @sseerrggii .

Esta PR añade las gemas necesarias para empezar a tener unos `spec/features/` con los que podamos garantizar que las features esencial y críticas de la app funcionan. 

De momento solo añado un test para login y otro para logout. Aún tengo que averigar como hacer tests ejecutando JS.

### CODE REVIEW

File: spec/features/sign_in_spec.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+describe 'sign in' do
+  let(:user) do
+    Fabricate(:user, email: 'user@timeoverflow.org', password: 'papapa22')
+  end
+
+  context 'with a valid password' do
+    it 'signs the user in' do
+      sign_in_with(user.email, 'papapa22')
+      expect(page).to have_content(I18n.t('application.navbar.sign_out'))
```
Comment (enricostano) [inline spec/features/sign_in_spec.rb]: `.sign_out`?

File: spec/features/transfer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  end
+
+  def submit_transfer_form_with(hours: nil, minutes: nil)
+    within transfer_form do
+      fill_in 'transfer_hours', with: hours
+      fill_in 'transfer_minutes', with: minutes
+
+      # hack alert! there is no translation for this string. How we build the
+      # copy then?
+      click_button 'Crear Transferencia'
```
Comment (enricostano) [inline spec/features/transfer_spec.rb]: In which language are we executing feature specs? We should use only one, english IMHO.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -8,9 +8,26 @@
 require 'capybara/rails'
 require 'database_cleaner'
 require 'fabrication'
+require 'selenium/webdriver'
```
Comment (enricostano) [inline spec/spec_helper.rb]: Do we need this for unit tests too?

File: spec/features/transfer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    Fabricate(:user, email: 'other_user@timeoverflow.org', password: 'papapa22')
+  end
+  let(:organization) { Fabricate(:organization) }
+
+  before do
+    # Create terms and conditions
+    Document.create(label: "t&c") do |doc|
+      doc.title = "Terms and Conditions"
+      doc.content = "blah blah blah"
+    end
```
Comment (enricostano) [inline spec/features/transfer_spec.rb]: the T&C in `spec_helper` are not enough?

Comment (sauloperez) [inline spec/features/transfer_spec.rb]: Yep. I fixed it with the last push and now I need to remove this.

File: spec/features/sign_in_spec.rb (hunk 1 above)
Comment (sauloperez) [inline spec/features/sign_in_spec.rb]: it checks that the log out link is present. That means we succesfully logged in.

File: spec/features/transfer_spec.rb (hunk 2 above)
Comment (sauloperez) [inline spec/features/transfer_spec.rb]: I haven't touched any language setting so I guess some sort of default.

File: spec/spec_helper.rb (hunk 3 above)
Comment (sauloperez) [inline spec/spec_helper.rb]: no, and `capybara/rails` either but where do you want to require them, in a `before(:all)`?

File: spec/features/sign_in_spec.rb (hunk 1 above)
Comment (enricostano) [inline spec/features/sign_in_spec.rb]: a bit misleading IMHO, maybe we should specify that in the example description or find another way to be check that the user is logged in

File: spec/spec_helper.rb (hunk 3 above)
Comment (enricostano) [inline spec/spec_helper.rb]: I used to create a `feature_spec_helper.rb` with this kind of stuff, but if I'm OK if we leave it here.

**Comment by sauloperez** [comment]
I give up, at least for now. Feature testing with angular it's painful and although [some people succeeded](https://forum.upcase.com/t/tdding-a-spa-application/4020/4) it appears not to be a common approach.

The main problem is that due to my lack of knowledge of angular, I can't manage to wait for it to "compile" the views. The DOM elements are still hidden when the test is executed and Capybara doesn't like that. Even forcing it to find hidden elements, I get `Selenium::WebDriver::Error::ElementNotVisibleError` errors.

Once we remove angular (I think it brings little to no value now) from the app we will be in a much better position.

**Comment by enricostano** [comment]
😭

**Comment by Morantron** [comment]
Fixing provisioning issues here https://github.com/coopdevs/timeoverflow-provisioning/pull/40

And adding chromedriver to devenv here

I'm now investigating DatabaseCleaner strategies. Looks like it's cleaning too compulsively :trollface: By the time the sign in request reaches the controller, the user is gone from DB :boom: Tried removing DatabaseCleaner and the sign in spec _works_ ( assertion needs to be fixed )

See screenshot from capybara here https://github.com/coopdevs/timeoverflow-provisioning/pull/41

wtf

**Comment by sauloperez** [comment]
🙈 we're pretty late to the capybara party so there must be some documentation to play it well with database cleaner.

**Comment by Morantron** [comment]
okiii, let's continue in #333 :dancer:

**Pull request closed without merging**